component/authenticator/src/main/java/org/wso2/carbon/identity/application/authenticator/totp/util/TOTPUtil.java

```diff
@@ -108,6 +108,14 @@ public class TOTPUtil {
      */
     public static String encrypt(String plainText) throws CryptoException {
 
+        String tenantDomain = CarbonContext.getThreadLocalCarbonContext().getTenantDomain();
+        // Get custom key from server configuration.
+        String customKey =
+                ServerConfiguration.getInstance().getFirstProperty(TOTP_KEY);
+        if (SUPER_TENANT_DOMAIN.equals(tenantDomain) && StringUtils.isNotBlank(customKey)) {
+            return CryptoUtil.getDefaultCryptoUtil().encryptWithCustomKeyAndBase64Encode(
```

Comment:
We should be able to refactor this if we change the kernel as mentioned there. Refer kernel PR for more details